src/org/purefn/starman/jedis.clj

```diff
@@ -29,18 +66,33 @@
         (random-sleep ms)
         (recur (inc cnt) (backoff ms))))))
 
+(defn- get* ^bytes
+  [^Jedis c ^String k]
+  (.get c (.getBytes k)))
+
+(defn- get-decoded
+  [^Jedis c fk encoder]
+  (some->> (get* c fk) (decode encoder)))
+
+(defn- set*
+  [^Jedis c k v]
+  (if (bytes? v)
```

Comment:
A little baffled by this... if v is a bytearray then k also needs to be one?

Reply:
yeah I was pretty baffled by this too, but I couldn't find an implementation of `set` where the key was a string and the value was a bytearray. There was only string/string or bytes/bytes